pkg/templates/notificationtemplates/notification_generator.go

```diff
@@ -1 +1,90 @@
 package notificationtemplates
+
+import (
+	"strings"
+
+	"github.com/codeready-toolchain/host-operator/pkg/templates/assets"
+
+	"github.com/pkg/errors"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("notification-templates")
+var templates map[string]NotificationTemplate
+
+// NotificationTemplate contains the template subject and content
+type NotificationTemplate struct {
+	Subject string
+	Content string
+}
+
+// Option is a function to set assets
+type Option func(asset *assets.Assets)
+
+// WithAssets is used to override the default assets
+func WithAssets(a assets.Assets) Option {
+	return func(assets *assets.Assets) {
+		templates = nil
+		assets.Names = a.Names
+		assets.Asset = a.Asset
+	}
+}
+
+// WithEmptyCache clears the cache of templates
+func WithEmptyCache() Option {
+	return func(_ *assets.Assets) {
+		templates = nil
+	}
+}
+
+// GetNotificationTemplates returns a notification subject, body and a boolean
+// indicating whether or not a template was found. Otherwise, an error will be returned
+func GetNotificationTemplates(name string, opts ...Option) (*NotificationTemplate, bool, error) {
```

Comment:
I suggest renaming this to `GetNotificationTemplate`, as it's returning a singular template.

Reply:
done here: 775cd43fa80a0e0d0238aa4dd9c5b9bd44a51745